avocado/core/test.py

```diff
@@ -572,17 +573,18 @@ def _run_avocado(self):
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
             raise exceptions.TestSkipError(details)
         except exceptions.TestCancel as details:
-            stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
-            skip_illegal_msg = ('Calling cancel() in setUp() '
-                                'is not allowed in avocado, you '
-                                'must fix your test. Original cancel exception: '
-                                '%s' % details)
-            raise exceptions.TestError(skip_illegal_msg)
+            cancel_test = details
         except:  # Old-style exceptions are not inherited from Exception()
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
             details = sys.exc_info()[1]
             raise exceptions.TestSetupFail(details)
         try:
+            #TODO: This should be removed when this try/testMethod()
+            #      becomes an else/testMethod(). Then the TestCance
+            #      exception has to be raised in the setUp() `except`.
```

Comment:
... which means no actions are required, just leaving a message that the comment was probably not really essential...